Automattic/pocket-casts-android
# Refactor downloads deep linking

## Description

> [!note]
> This will be a series of PRs that move deep linking to a separate module that can be tested. The goal is to fix #2405 but I don't want to do it blindly without having tests in order to not introduce regressions to deep linking.

This PR migrates deep linking of downloads to the new module.

## Testing Instructions

1. Start downloading some episodes.
2. While download notification is showing tap on it.
3. The downloads page should open.

## Checklist
